diff -up net-tools-1.60/lib/interface.c.trim-iface net-tools-1.60/lib/interface.c
--- net-tools-1.60/lib/interface.c.trim-iface	2011-12-02 18:00:09.000000000 +0100
+++ net-tools-1.60/lib/interface.c	2011-12-02 18:02:50.143689438 +0100
@@ -638,7 +638,7 @@ int do_if_print(struct interface *ife, v
 
 void ife_print_short(struct interface *ptr)
 {
-    printf("%-9.9s ", ptr->name);
+    printf("%-9s ", ptr->name);
     printf("%5d %3d ", ptr->mtu, ptr->metric);
     /* If needed, display the interface statistics. */
     if (ptr->statistics_valid) {
