diff -up net-tools-1.60/lib/inet6_gr.c.scanf-format net-tools-1.60/lib/inet6_gr.c
--- net-tools-1.60/lib/inet6_gr.c.scanf-format	2011-12-06 13:14:57.000000000 +0100
+++ net-tools-1.60/lib/inet6_gr.c	2011-12-06 13:21:01.417616990 +0100
@@ -82,7 +82,7 @@ int rprint_fib6(int ext, int numeric)
 	     "Flag Met Ref Use If\n"));
 
     while (fgets(buff, 1023, fp)) {
-	num = sscanf(buff, "%4s%4s%4s%4s%4s%4s%4s%4s %02x %4s%4s%4s%4s%4s%4s%4s%4s %02x %4s%4s%4s%4s%4s%4s%4s%4s %08x %08x %08x %08x %s\n",
+	num = sscanf(buff, "%4s%4s%4s%4s%4s%4s%4s%4s %02x %4s%4s%4s%4s%4s%4s%4s%4s %02x %4s%4s%4s%4s%4s%4s%4s%4s %08x %08x %08x %08x %15s\n",
 		     addr6p[0], addr6p[1], addr6p[2], addr6p[3],
 		     addr6p[4], addr6p[5], addr6p[6], addr6p[7],
 		     &prefix_len,
diff -up net-tools-1.60/lib/inet_gr.c.scanf-format net-tools-1.60/lib/inet_gr.c
--- net-tools-1.60/lib/inet_gr.c.scanf-format	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/lib/inet_gr.c	2011-12-06 13:23:25.408816852 +0100
@@ -68,7 +68,7 @@ int rprint_fib(int ext, int numeric)
     mss = 0;
 
     fmt = proc_gen_fmt(_PATH_PROCNET_ROUTE, 0, fp,
-		       "Iface", "%16s",
+		       "Iface", "%15s",
 		       "Destination", "%127s",
 		       "Gateway", "%127s",
 		       "Flags", "%X",
@@ -80,7 +80,7 @@ int rprint_fib(int ext, int numeric)
 		       "Window", "%d",
 		       "IRTT", "%d",
 		       NULL);
-    /* "%16s %127s %127s %X %d %d %d %127s %d %d %d\n" */
+    /* "%15s %127s %127s %X %d %d %d %127s %d %d %d\n" */
 
     if (!fmt)
 	return 1;
@@ -268,7 +268,7 @@ int rprint_cache(int ext, int numeric)
 		 "MSS   Window irtt  HH  Arp\n"));
 
       fmt = proc_gen_fmt(_PATH_PROCNET_RTCACHE, 0, fp,
-		       "Iface", "%16s",
+		       "Iface", "%15s",
 		       "Destination", "%127s",
 		       "Gateway", "%127s",
 		       "Flags", "%X",
@@ -282,7 +282,7 @@ int rprint_cache(int ext, int numeric)
 		       "HH", "%d",
 		       "ARP", "%d",
 		       NULL);
-      /* "%16s %127s %127s %X %d %d %d %127s %d %d %d %d %d\n" */
+      /* "%15s %127s %127s %X %d %d %d %127s %d %d %d %d %d\n" */
     }
 
     if (format == 2) {
@@ -291,7 +291,7 @@ int rprint_cache(int ext, int numeric)
 		 "Flags Metric Ref    Use Iface    "
 		 "MSS   Window irtt  TOS HHRef HHUptod     SpecDst\n"));
         fmt = proc_gen_fmt(_PATH_PROCNET_RTCACHE, 0, fp,
-		       "Iface", "%16s",
+		       "Iface", "%15s",
 		       "Destination", "%127s",
 		       "Gateway", "%127s",
 		       "Flags", "%X",
@@ -307,7 +307,7 @@ int rprint_cache(int ext, int numeric)
 		       "HHUptod", "%d",
 		       "SpecDst", "%127s",
 		       NULL);
-      /* "%16s %127s %127s %X %d %d %d %127s %d %d %d %d %d %127s\n" */
+      /* "%15s %127s %127s %X %d %d %d %127s %d %d %d %d %d %127s\n" */
     }
 
 
