diff -up net-tools-1.60/lib/ib.c.ib-warning net-tools-1.60/lib/ib.c
--- net-tools-1.60/lib/ib.c.ib-warning	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/lib/ib.c	2011-12-06 13:40:32.463976896 +0100
@@ -47,7 +47,10 @@ static const char *pr_ib(const char *ptr
 	pos += sprintf(pos, "%02X:", (*ptr++ & 0377));
     }
     buff[strlen(buff) - 1] = '\0';
-
+    char *ib_warning = "Ifconfig uses the ioctl access method to get the full address information, which limits hardware addresses to 8 bytes.\n"
+                       "Because Infiniband address has 20 bytes, only the first 8 bytes are displayed correctly.\n"
+                       "Ifconfig is obsolete! For replacement check ip.\n";
+    fprintf(stderr, _(ib_warning));
     /* snprintf(buff, sizeof(buff), "%02X:%02X:%02X:%02X:%02X:%02X",
 	     (ptr[0] & 0377), (ptr[1] & 0377), (ptr[2] & 0377),
 	     (ptr[3] & 0377), (ptr[4] & 0377), (ptr[5] & 0377)
diff -up net-tools-1.60/man/en_US/ifconfig.8.ib-warning net-tools-1.60/man/en_US/ifconfig.8
--- net-tools-1.60/man/en_US/ifconfig.8.ib-warning	2011-12-06 13:40:31.000000000 +0100
+++ net-tools-1.60/man/en_US/ifconfig.8	2011-12-06 13:41:20.467376771 +0100
@@ -221,6 +221,12 @@ for more information.
 .br
 .I /proc/net/if_inet6
 .SH BUGS
+Ifconfig uses obsolete kernel interface.
+It uses the ioctl access method to get the full address information,
+which limits hardware addresses to 8 bytes.
+Since an Infiniband address is 20 bytes,
+only the first 8 bytes of Infiniband address are displayed.
+.LP
 While appletalk DDP and IPX addresses will be displayed they cannot be
 altered by this command.
 .SH SEE ALSO
