diff -up net-tools-1.60/ifconfig.c.IA64 net-tools-1.60/ifconfig.c
--- net-tools-1.60/ifconfig.c.IA64	2011-12-06 14:36:57.158662389 +0100
+++ net-tools-1.60/ifconfig.c	2011-12-06 14:36:57.247661276 +0100
@@ -914,7 +914,8 @@ int main(int argc, char **argv)
 		    continue;
 		}
 		
-		memcpy(&ip, &sin.sin_addr.s_addr, sizeof(unsigned long));
+		memset(&ip, 0, sizeof(unsigned long));
+		memcpy(&ip, &sin.sin_addr.s_addr, sizeof(sin.sin_addr.s_addr));
 		
 		if (get_nmbc_parent(ifr.ifr_name, &nm, &bc) < 0) {
 		    fprintf(stderr, _("Interface %s not initialized\n"),
