diff -up net-tools-1.60/lib/pathnames.h.x25 net-tools-1.60/lib/pathnames.h
--- net-tools-1.60/lib/pathnames.h.x25	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/lib/pathnames.h	2011-12-05 17:00:49.917886688 +0100
@@ -49,8 +49,8 @@
 #define _PATH_PROCNET_RARP		"/proc/net/rarp"
 #define _PATH_ETHERS			"/etc/ethers"
 #define _PATH_PROCNET_ROSE_ROUTE	"/proc/net/rose_routes"
-#define _PATH_PROCNET_X25              "/proc/net/x25"
-#define _PATH_PROCNET_X25_ROUTE                "/proc/net/x25_routes"
+#define _PATH_PROCNET_X25		"/proc/net/x25"
+#define _PATH_PROCNET_X25_ROUTE		"/proc/net/x25/route"
 #define _PATH_PROCNET_DEV_MCAST		"/proc/net/dev_mcast"
 #define _PATH_PROCNET_ATALK_ROUTE	"/proc/net/atalk_route"
 
