diff -up net-tools-1.60/lib/interface.c.virtualname net-tools-1.60/lib/interface.c
--- net-tools-1.60/lib/interface.c.virtualname	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/lib/interface.c	2011-11-23 18:54:31.881194851 +0100
@@ -597,8 +597,8 @@ int do_if_print(struct interface *ife, v
 
 void ife_print_short(struct interface *ptr)
 {
-    printf("%-5.5s ", ptr->name);
-    printf("%5d %-2d ", ptr->mtu, ptr->metric);
+    printf("%-9.9s ", ptr->name);
+    printf("%5d %3d ", ptr->mtu, ptr->metric);
     /* If needed, display the interface statistics. */
     if (ptr->statistics_valid) {
 	printf("%8llu %6lu %6lu %-6lu ",
@@ -608,7 +608,7 @@ void ife_print_short(struct interface *p
 	       ptr->stats.tx_packets, ptr->stats.tx_errors,
 	       ptr->stats.tx_dropped, ptr->stats.tx_fifo_errors);
     } else {
-	printf("%-56s", _("     - no statistics available -"));
+	printf("%-60s", _("     - no statistics available -"));
     }
     /* DONT FORGET TO ADD THE FLAGS IN ife_print_long, too */
     if (ptr->flags == 0)
diff -up net-tools-1.60/netstat.c.virtualname net-tools-1.60/netstat.c
--- net-tools-1.60/netstat.c.virtualname	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/netstat.c	2011-11-23 18:54:00.563586374 +0100
@@ -1619,7 +1619,7 @@ static int iface_info(void)
     }
     if (flag_exp < 2) {
 	ife_short = 1;
-	printf(_("Iface   MTU Met   RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
+	printf(_("Iface       MTU Met    RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
     }
 
     if (for_all_interfaces(do_if_print, &flag_all) < 0) {
